server/app/models/host_node.rb

```diff
@@ -45,9 +47,10 @@ class HostNode
   after_save :reserve_node_number, :ensure_unique_name
 
   index({ grid_id: 1 })
-  index({ node_id: 1 })
+  index({ node_id: 1 }, { unique: true, sparse: true })
```

Comment:
Do these kinds of changes to indexes actually get applied to the mongodb, or does this need a migration?